audit/audit.go

```diff
@@ -0,0 +1,72 @@
+package audit
+
+import (
+	"context"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// CryptoNewKeyEvent occurs when creating a new key.
+	CryptoNewKeyEvent = "CreateNewKey"
+	// CryptoSignJWTEvent occurs when signing a JWT.
+	CryptoSignJWTEvent = "SignJWT"
+	// CryptoSignJWSEvent occurs when signing a JWS.
+	CryptoSignJWSEvent = "SignJWS"
+	// CryptoDecryptEvent occurs when decrypting data.
+	CryptoDecryptEvent = "Decrypt"
+)
+
+// Info provides contextual information for auditable events.
+type Info struct {
+	// Actor is the user or service that performed the operation.
+	Actor string
+	// Operation is the name of the operation that was performed.
+	Operation string
+}
+
+// NewInfo formats audit info
+func NewInfo(actor string, operationModuleName string, operation string) Info {
+	return Info{
+		Actor:     actor,
+		Operation: operationModuleName + "." + operation,
+	}
+}
+
+var auditInfoContextKey = struct{}{}
+
+// Context returns a child context of the given parent context, enriched with the auditable actor and performed operation.
+func Context(parent context.Context, provider func() Info) context.Context {
```

Comment:
Because the audit middleware gets invoked before the API's preprocessors/middleware, which (the latter) sets the module and operation name, which I want to be part of the audit log. But it is not set yet when the audit middleware is invoked.

I realized this can be solved by invoking the audit middleware from the API layer, then it can be just `Info` again instead of a func.